usecases/auth/authentication/apikey/db_users.go

```diff
@@ -197,7 +243,102 @@ func (c *DBUser) validateStrongHash(key, secureHash, userId string) error {
 		return fmt.Errorf("invalid token")
 	}
 	token := []byte(key + secureHash)
-	c.weakKeyStorageById[userId] = sha256.Sum256(token)
+	c.memoryOnyData.WeakKeyStorageById[userId] = sha256.Sum256(token)
+
+	return nil
+}
+
+func (c *DBUser) Snapshot() DBUserSnapshot {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	return DBUserSnapshot{Data: c.data, Version: SnapshotVersion}
+}
+
+func (c *DBUser) Restore(snapshot DBUserSnapshot) error {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	if snapshot.Version != SnapshotVersion {
+		return fmt.Errorf("invalid snapshot version")
+	}
+	c.data = snapshot.Data
 
 	return nil
 }
+
+func (c *DBUser) storeToFile() error {
+	data, err := json.Marshal(DBUserSnapshot{Data: c.data, Version: SnapshotVersion})
+	if err != nil {
+		return err
+	}
+
+	tmpFile, err := os.CreateTemp(c.path, "temp-*.tmp")
```

Comment:
this part will create a temp file named `"temp-*.tmp"`? with asterisk in it?

Reply:
The docstring says:

> The filename is generated by taking pattern and adding a random string to the end. If pattern includes a "*", the random string replaces the last "*"